src/microbe_stage/CompoundCloudSystem.cs

```diff
@@ -332,33 +330,25 @@ public void GetAllAvailableAt(Vector3 worldPosition, Dictionary<Compound, float>
             // relative origin point is calculated and that is restricted by
             // checking if the point is within the circle before grabbing
 
-            // And apparently there isn't an algorithm to generate the points with closest ones first? so
-            // this goes through everything and keeps the closest found point
-
-            int xEnd = (int)Mathf.Round(cloudRelativeX + localRadius);
-            int yEnd = (int)Mathf.Round(cloudRelativeY + localRadius);
+            // TODO: pull these to a constant to not constantly recalculate them
+            double fullCircle = Math.PI * 2;
+            double searchArc = fullCircle / 24.0;
 
-            for (int x = (int)Mathf.Round(cloudRelativeX - localRadius);
-                 x <= xEnd;
-                 x += 1)
+            for (int radious = 1;
+                 radious < localRadius;
+                 radious += 1)
             {
-                for (int y = (int)Mathf.Round(cloudRelativeY - localRadius);
-                     y <= yEnd;
-                     y += 1)
+                for (double theta = 0;
```

Comment:
Are doubles *actually* needed or would this work with floats? I guess either way this won't have much of an effect (maybe) as the float variant of math methods just convert to and back from doubles anyway...

Reply:
When dividing 2 pi by 24 (and someone may up that granularity in the future), then doing trig out at long distances, I think floats will start making some trouble.

Reply:
I suppose this is better to use it this way, then.